sdk/include/opentelemetry/sdk/common/attribute_utils.h

```diff
@@ -114,19 +115,32 @@ class AttributeMap
         });
   }
 
+  // Construct map from initializer list by applying `SetAttribute` transform for every attribute
+  AttributeMap(
+      std::initializer_list<std::pair<nostd::string_view, opentelemetry::common::AttributeValue>>
+          attributes)
+      : AttributeMap()
+  {
+    for (auto &kv : attributes)
+    {
+      SetAttribute(kv.first, kv.second);
+    }
+  }
+
+  // Returns a reference to this map
   const std::unordered_map<std::string, OwnedAttributeValue> &GetAttributes() const noexcept
   {
-    return attributes_;
+    return (*this);
   }
 
+  // Convert non-owning key-value to owning std::string(key) and OwnedAttributeValue(value)
   void SetAttribute(nostd::string_view key,
                     const opentelemetry::common::AttributeValue &value) noexcept
   {
-    attributes_[std::string(key)] = nostd::visit(converter_, value);
+    (*this)[std::string(key)] = nostd::visit(converter_, value);
   }
 
 private:
-  std::unordered_map<std::string, OwnedAttributeValue> attributes_;
```

Comment:
I removed internal `std::unordered_map` here. It is much nicer to keep all characteristics of `unordered_map` retained on `AttributeMap` object itself, i.e. you can now range-for-loop iterate over that, etc.. I see this as a reusable helper class we can consume within the SDK more often.